===================================================================
RCS file: /c/ncvs/src/crypto/openssh/session.c,v
retrieving revision 1.4.2.10
retrieving revision 1.4.2.11
diff -u -p -r1.4.2.10 -r1.4.2.11
--- src/crypto/openssh/session.c	2001/11/21 10:45:15	1.4.2.10
+++ src/crypto/openssh/session.c	2001/12/03 00:53:28	1.4.2.11
@@ -1154,6 +1154,7 @@ do_child(Session *s, const char *command
 		child_set_env(&env, &envsize, "TZ", getenv("TZ"));
 
 	/* Set custom environment options from RSA authentication. */
+	if (!options.use_login) {
 	while (custom_environment) {
 		struct envstring *ce = custom_environment;
 		char *s = ce->s;
@@ -1166,6 +1167,7 @@ do_child(Session *s, const char *command
 		custom_environment = ce->next;
 		xfree(ce->s);
 		xfree(ce);
+	}
 	}
 
 	snprintf(buf, sizeof buf, "%.50s %d %d",
